src/track.c

```diff
@@ -676,12 +675,18 @@ void calc_navigation_measurement_(u8 n_channels, channel_measurement_t* meas[],
     nav_meas[i]->carrier_phase += (nav_time - meas[i]->receiver_time) * meas[i]->carrier_freq;
 
     nav_meas[i]->lock_counter = meas[i]->lock_counter;
+	
+	// calc the clock error
+	calc_sat_state(ephemerides[i], nav_meas[i]->tot,
+                   nav_meas[i]->sat_pos, nav_meas[i]->sat_vel,
+                   &clock_err, &clock_rate_err);
```

Comment:
I think you have look algorithmically outside of this code as to the effect
of selecting and remaining on a specific ordinal satellite. Not sure I saw
any elevation masking code, or lets stick with this satellite until it
reaches X degrees of elevation, type code.

I targeted finding a suitably high one so the transition between visible
satellites has the least impact on the numbers, rather than explicitly
defining that it needs to be the *highest*.

Ken's observation is that all the RTK solution jumping (decamping) is
around the time the satellite which currently gets picked fades in and out
of visibility and lock. Mine more generally is that rate of change (clock
drift) on the receiver is generally fairly consistent short of someone
blowing on the XO/TCXO (which causes massive acceleration/deceleration
events to befall the correlators). I'm bringing my own baggage to the party
here, as I don't have a sufficiently clear view of the software/hardware in
this case. I'll talk with Fergus/Colin about this in another venue. My
other observation is there is a lot of math here that's done over and over,
so there's less opportunity to view an epoch-to-epoch change.

I'm tinkering with other changes to separate the received observations from
the common set used for the RTK solution. The buffers for RTK or matrix
rise to the square so you're pretty much memory locked to 11 channels
(perhaps eek 12), when realistically you could get 18 measurements from a
base (sub 256 byte packet, not sure I ever see more than 14), and get down
to 8 or so good candidates for the RTK side, ie good sustained cycle lock
and higher mask angle.

On Thu, Apr 30, 2015 at 11:58 AM, Fergus Noble <notifications@github.com>
wrote:

> In src/track.c
> <https://github.com/swift-nav/libswiftnav/pull/138#discussion_r29447882>:
>
> > @@ -676,12 +675,18 @@ void calc_navigation_measurement_(u8 n_channels, channel_measurement_t* meas[],
> >      nav_meas[i]->carrier_phase += (nav_time - meas[i]->receiver_time) * meas[i]->carrier_freq;
> >
> >      nav_meas[i]->lock_counter = meas[i]->lock_counter;
> > +	
> > +	// calc the clock error
> > +	calc_sat_state(ephemerides[i], nav_meas[i]->tot,
> > +                   nav_meas[i]->sat_pos, nav_meas[i]->sat_vel,
> > +                   &clock_err, &clock_rate_err);
>
> Also, I believe the satellite clock error is small enough (<1% of
> pseudorange) that this effect isn't very significant, but if its possible
> to do it this way without additional overhead then it sounds good to me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/swift-nav/libswiftnav/pull/138/files#r29447882>.
>